src/drt/src/dr/FlexDR_init.cpp

```diff
@@ -1111,7 +1111,7 @@ void FlexDRWorker::initNet_termGenAp_new(const frDesign* design, drPin* dPin)
       }
     }
   } else if (dPinTerm->typeId() == frcTerm) {
-    auto term = static_cast<frTerm*>(dPinTerm);
+    auto term = static_cast<frMTerm*>(dPinTerm);
```

Comment:
the typeId and the class should match (frcTerm vs frMTerm)

Reply:
I initially avoided this because they are handled the same in all cases where term or pin type is considered. I kept the type identifier for `MTerm`/`BTerm` as `frcTerm` and the identifiers for `MPin`/`BPin` as `frcPin`. I can change the types to be more specific.